staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go

```diff
@@ -405,63 +417,17 @@ func (cnc *CloudNodeController) initializeNode(ctx context.Context, node *v1.Nod
 // a node object with provider-specific information.
 // All of the returned functions are idempotent, because they are used in a retry-if-conflict
 // loop, meaning they could get called multiple times.
-func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(ctx context.Context, node *v1.Node) ([]nodeModifier, error) {
-	var (
-		nodeModifiers []nodeModifier
-		providerID    string
-		err           error
-	)
+func (cnc *CloudNodeController) getNodeModifiersFromCloudProvider(
+	ctx context.Context,
+	node *v1.Node,
+	providerID string,
+	providerIDModifier nodeModifier,
```

Comment:
remove this and just pass in getProviderID directly below?

Reply:
See https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/cloud-provider/controllers/node/node_controller.go#L415-L436
We should handle providerID not set scenario and get this modifier before get instance metadata that's why I move this out and also pass providerID and related modifier.

Reply:
The modifer is very simple though:
```
modifier := func(n *v1.Node) {
    if n.Spec.ProviderID == "" {
        n.Spec.ProviderID = providerID
    }
}
````

We can define the modifer here if providerID is "".

Reply:
Thanks, looks good, simplified your suggestion and updated.